docs/source/daml2ts/index.rst

```diff
@@ -0,0 +1,291 @@
++-------------------+------------------+----------------------------------+---------------------------------------------------------------------+
+
+DAML to Typescript mappings
+---------------------------
+
+The mappings from DAML to Typescript are best explained by example.
+
+Records
+~~~~~~~
+
+Basic
+*****
+
+In DAML, we might model a person like this.
+
+.. code-block:: daml
+   :linenos:
+
+   data Person =
+     Person
+       name : Text
+       party : Party
+       age : Int
+
+Given the above definition, the generated Typescript code will be as follows.
+
+   .. code-block:: typescript
+      :linenos:
+
+      export type Person = {
+        name: string;
+        party: daml.Party;
+        age: daml.Int;
+      }
+
+      export const Person: daml.Serializable<Person> = ({
+        decoder: () => jtv.object({
+          name: daml.Text.decoder(),
+          party: daml.Party.decoder(),
+          aage: daml.Int.decoder(),
+        }),
+      })
+
+Generic
+*******
+
+In the above case, ``Person`` is a monomorphic type. If we change it slightly so that it is polymorphic in the type of ``age``, it would look like this.
+
+.. code-block:: daml
+   :linenos:
+
+   data Person a =
+     Person
+       name : Text
+       party : Party
+       age : a
+
+In this case, the generated Typescript would read like the following.
+
+.. code-block:: typescript
+   :linenos:
+
+   export type Person<a> = {
+     name: string;
+     party: daml.Party;
+     age: a;
+   }
+
+   export const Person = <a>(a: daml.Serializable<a>): daml.Serializable<Person<a>> => ({
+     decoder: () => jtv.object({
+       name: daml.Text.decoder(),
+       party: daml.Party.decoder(),
+       age: a.decoder(),
+     }),
+   })
+
+Tuples
+~~~~~~
+
+Suppose the following DAML definition.
+
+.. code-block:: daml
+   :linenos:
+
+   data Pair a b = Pair (a, b)
+
+In Typescript, the above definition renders as follows.
+
+.. code-block:: typescript
+   :linenos:
+
+   import * as pkg40f452260bef3f29dede136108fc08a88d5a5250310281067087da6f0baddff7_DA_Types from './../40f452260bef3f29dede136108fc08a88d5a5250310281067087da6f0baddff7/DA/Types';
+
+   export type Pair<a, b> =
+     |  { tag: 'Pair'; value: pkg40f452260bef3f29dede136108fc08a88d5a5250310281067087da6f0baddff7_DA_Types.Tuple2<a, b> }
+
+   export const Pair = <a, b>(a: daml.Serializable<a>, b: daml.Serializable<b>): daml.Serializable<Pair<a, b>> => ({
+     decoder: () => jtv.oneOf<Pair<a, b>>(
+       jtv.object({tag: jtv.constant('Pair'), value: jtv.lazy(() => pkg40f452260bef3f29dede136108fc08a88d5a5250310281067087da6f0baddff7_DA_Types.Tuple2(a, b).decoder())}),
+     ),
+   });
+
+What we are interested in here is the treatment of the tuple type ``(a, b)``. As the above shows, its presence has been captured by a type ``Tuple2<a, b>`` from a package generated from the DAML standard library. Here is the definition.
```

Comment:
It’s a bit unclear from this paragraph that this is not something that is specific to the typescript codegen but rather that this is just how the types are represented in DAML-LF.

Reply:
This is another detail that is going to get removed. We are just going to inline `{ _1: a, _2: b }` and do away with these technical details in this documentation altogether (except perhaps for a note explaining that we have omitted them).